homeassistant/components/template/weather.py

```diff
@@ -327,4 +367,66 @@ async def async_added_to_hass(self) -> None:
                 "_forecast",
                 self._forecast_template,
             )
+
+        if self._forecast_daily_template:
+            self.add_template_attribute(
+                "_forecast_daily",
+                self._forecast_daily_template,
+                on_update=self._update_daily,
+                validator=self._validate_forecast,
+            )
+        if self._forecast_hourly_template:
+            self.add_template_attribute(
+                "_forecast_hourly",
+                self._forecast_hourly_template,
+                on_update=self._update_hourly,
+                validator=self._validate_forecast,
+            )
+        if self._forecast_twice_daily_template:
+            self.add_template_attribute(
+                "_forecast_twice_daily",
+                self._forecast_twice_daily_template,
+                on_update=self._update_twice_daily,
+                validator=self._validate_forecast,
+            )
+
         await super().async_added_to_hass()
+
+    @callback
+    def _update_daily(self, result: list[Forecast] | TemplateError) -> None:
+        """Save template result and trigger forecast listener."""
+        attr_result = None if isinstance(result, TemplateError) else result
+        setattr(self, "_forecast_daily", attr_result)
+        self.hass.create_task(self.async_update_listeners(["daily"]))
+
+    @callback
+    def _update_hourly(self, result: list[Forecast] | TemplateError) -> None:
+        """Save template result and trigger forecast listener."""
+        attr_result = None if isinstance(result, TemplateError) else result
+        setattr(self, "_forecast_hourly", attr_result)
+        self.hass.create_task(self.async_update_listeners(["hourly"]))
+
+    @callback
+    def _update_twice_daily(self, result: list[Forecast] | TemplateError) -> None:
+        """Save template result and trigger forecast listener."""
+        attr_result = None if isinstance(result, TemplateError) else result
+        setattr(self, "_forecast_twice_daily", attr_result)
+        self.hass.create_task(self.async_update_listeners(["twice_daily"]))
+
+    @callback
+    def _validate_forecast(
+        self, result: list[Forecast] | TemplateError
```

Comment:
The result is not yet known to be a list of forecasts, that's what's being validated?

Reply:
Not sure what you mean here?
If result is empty we return `None`
If template error we return `None`
If other validation fails we raise `vol.Invalid`